src/lib.rs

```diff
@@ -66,8 +66,8 @@ pub unsafe trait IsArray {
 #[inline]
 /// Initialize an array given an initializer expression.
 ///
-/// The initializer is given the index of the element. It is allowed
-/// to mutate external state; we will always initialize the elements in order.
+/// The initializer is given the index (between 0 and `Array:len() - 1` included) of the element. It is allowed
```

Comment:
I don't think this commit is particularly helpful. "In order" should suffice, there is no need to explain the bounds

Reply:
Agreed